app/scripts/controllers/network/provider-api-tests/shared-tests.js

```diff
@@ -823,22 +790,16 @@ export function testsForRpcMethodSupportingBlockParam(
         comms.mockNextBlockTrackerRequest({ blockNumber: '0x100' });
         // The block-ref middleware will make the request as specified
         // except that the block param is replaced with the latest block
-        // number.
+        // number, and we delay it.
         comms.mockInfuraRpcCall({
+          delay: 100,
           request: buildRequestWithReplacedBlockParam(
             requests[0],
             blockParamIndex,
             '0x100',
           ),
           response: { result: mockResults[0] },
         });
-        // This is the original request as below, which the block-ref
-        // middleware will allow through, except that we delay it.
-        comms.mockInfuraRpcCall({
-          request: requests[0],
-          response: { result: mockResults[0] },
-          delay: 100,
```

Comment:
I think it's important that we delay one of these requests to prevent this test from failing intermittently. This test is different from other tests in that we issue multiple requests at the exact same time. We want the first one (the one with the first result) to be processed first, and then the other ones afterward. So the delay should guarantee the right result.